src/AbstractMenu.js

```diff
@@ -0,0 +1,117 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+
+import MenuItem from './MenuItem';
+
+export default class AbstractMenu extends Component {
+    static propTypes = {
+        children: PropTypes.node.isRequired
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.seletedItemRef = null;
+        this.state = {
+            selectedItem: null,
+            forceSubMenuOpen: false
+        };
+    }
+
+    handleKeyNavigation = (e) => {
+        switch (e.keyCode) {
+            case 37: // left arrow
+            case 27: // escape
+                e.preventDefault();
+                this.hideMenu(e);
+                break;
+            case 38: // up arrow
+                e.preventDefault();
+                this.selectChildren(true);
+                break;
+            case 40: // down arrow
+                e.preventDefault();
+                this.selectChildren(false);
+                break;
+            case 39: // right arrow
+                this.tryToOpenSubMenu(e);
+                break;
+            case 13: // enter
+                e.preventDefault();
+                this.tryToOpenSubMenu(e);
+                if (this.seletedItemRef && this.seletedItemRef.ref) this.seletedItemRef.ref.click();
```

Comment:
Because of "Illegal invocation" something like this will be needed `if (this.seletedItemRef && this.seletedItemRef.ref && this.seletedItemRef.ref.click) callIfExists(this.seletedItemRef.ref.click.bind(this.seletedItemRef.ref))`.
I think it's better to check that `this.seletedItemRef.ref` is an instance of `Element`. Every `HTMLElement` has a valid click function (https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement)